## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 434303`

Logs: https://github.com/ttrssreal/nixpkgs-review-gha/actions/runs/17020159708


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>angle</li>
    <li>ladybird</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>angle</li>
    <li>ladybird</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>angle</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>angle</summary>
<pre>Warning: supplying the --target x86_64-apple-macos != x86_64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[789/1183] CXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/vk_resource.o
Warning: supplying the --target x86_64-apple-macos != x86_64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[790/1183] CXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/vk_utils.o
Warning: supplying the --target x86_64-apple-macos != x86_64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[791/1183] OBJCXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/DisplayVkMac.o
Warning: supplying the --target x86_64-apple-macos != x86_64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[792/1183] AR obj/libangle_image_util.a
[793/1183] AR obj/third_party/vulkan-loader/src/libvulkan.a
[794/1183] OBJCXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/IOSurfaceSurfaceVkMac.o
Warning: supplying the --target x86_64-apple-macos != x86_64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[795/1183] SOLINK libVkICD_mock_icd.dylib libVkICD_mock_icd.dylib.TOC
FAILED: libVkICD_mock_icd.dylib libVkICD_mock_icd.dylib.TOC 
TOOL_VERSION=1755428393 ../../build/toolchain/apple/linker_driver.py -Wcrl,otoolpath,/nix/store/cv9mkbgidzw7hmk1mfr5dj6f73z7din1-apple-sdk-15.2/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool -Wcrl,nmpath,/nix/store/cv9mkbgidzw7hmk1mfr5dj6f73z7din1-apple-sdk-15.2/Toolchains/XcodeDefault.xctoolchain/usr/bin/nm -Wcrl,tocname,"./libVkICD_mock_icd.dylib.TOC" -Wcrl,driver,../../../../../store/zzwkn6pg7npgnxwwzxad9z5vzizc2i2x-angle-clang-llvm-join/bin/clang++ -Wcrl,strippath,/nix/store/cv9mkbgidzw7hmk1mfr5dj6f73z7din1-apple-sdk-15.2/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip -Wcrl,installnametoolpath,../../../../../store/zzwkn6pg7npgnxwwzxad9z5vzizc2i2x-angle-clang-llvm-join/bin/llvm-install-name-tool -shared -Wl,--no-warn-duplicate-rpath -Wl,-no_warn_duplicate_libraries -fuse-ld=lld -Wl,-fatal_warnings -Wl,--icf=all -Wl,--color-diagnostics -Wl,--strict-auto-link -Wl,--no-call-graph-profile-sort --target=x86_64-apple-macos -no-canonical-prefixes -Wl,-dead_strip -nostdlib++ -isysroot ../../../../../store/cv9mkbgidzw7hmk1mfr5dj6f73z7din1-apple-sdk-15.2/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacos-version-min=12.0 -Wl,-ObjC -o "./libVkICD_mock_icd.dylib" "@./libVkICD_mock_icd.dylib.rsp" 
/bin/sh: ../../build/toolchain/apple/linker_driver.py: /usr/bin/env: bad interpreter: Operation not permitted
[796/1183] OBJCXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/WindowSurfaceVkMac.o
Warning: supplying the --target x86_64-apple-macos != x86_64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[797/1183] CXX obj/third_party/vulkan_memory_allocator/vulkan_memory_allocator/vma.o
Warning: supplying the --target x86_64-apple-macos != x86_64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
ninja: build stopped: subcommand failed.</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>angle</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>angle</summary>
<pre>Warning: supplying the --target arm64-apple-macos != arm64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[787/1182] CXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/vk_resource.o
Warning: supplying the --target arm64-apple-macos != arm64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[788/1182] CXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/vk_renderer.o
Warning: supplying the --target arm64-apple-macos != arm64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[789/1182] CXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/vk_utils.o
Warning: supplying the --target arm64-apple-macos != arm64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[790/1182] OBJCXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/DisplayVkMac.o
Warning: supplying the --target arm64-apple-macos != arm64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[791/1182] AR obj/libangle_image_util.a
[792/1182] AR obj/third_party/vulkan-loader/src/libvulkan.a
[793/1182] SOLINK libVkICD_mock_icd.dylib libVkICD_mock_icd.dylib.TOC
FAILED: libVkICD_mock_icd.dylib libVkICD_mock_icd.dylib.TOC 
TOOL_VERSION=1755428284 ../../build/toolchain/apple/linker_driver.py -Wcrl,otoolpath,/nix/store/zm6nfi7l8yq7s0i3m52pxj00nr3n76j5-apple-sdk-15.2/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool -Wcrl,nmpath,/nix/store/zm6nfi7l8yq7s0i3m52pxj00nr3n76j5-apple-sdk-15.2/Toolchains/XcodeDefault.xctoolchain/usr/bin/nm -Wcrl,tocname,"./libVkICD_mock_icd.dylib.TOC" -Wcrl,driver,../../../../../store/r9m4n9l3wz6rdzy6l0xs01fzl82iwmiy-angle-clang-llvm-join/bin/clang++ -Wcrl,strippath,/nix/store/zm6nfi7l8yq7s0i3m52pxj00nr3n76j5-apple-sdk-15.2/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip -Wcrl,installnametoolpath,../../../../../store/r9m4n9l3wz6rdzy6l0xs01fzl82iwmiy-angle-clang-llvm-join/bin/llvm-install-name-tool -shared -Wl,--no-warn-duplicate-rpath -Wl,-no_warn_duplicate_libraries -fuse-ld=lld -Wl,-fatal_warnings -Wl,--icf=all -Wl,--color-diagnostics -Wl,--strict-auto-link -Wl,--no-call-graph-profile-sort --target=arm64-apple-macos -Wl,-mllvm,-enable-machine-outliner=never -no-canonical-prefixes -Wl,-dead_strip -nostdlib++ -isysroot ../../../../../store/zm6nfi7l8yq7s0i3m52pxj00nr3n76j5-apple-sdk-15.2/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacos-version-min=12.0 -Wl,-ObjC -o "./libVkICD_mock_icd.dylib" "@./libVkICD_mock_icd.dylib.rsp" 
/bin/sh: ../../build/toolchain/apple/linker_driver.py: /usr/bin/env: bad interpreter: Operation not permitted
[794/1182] OBJCXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/IOSurfaceSurfaceVkMac.o
Warning: supplying the --target arm64-apple-macos != arm64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
[795/1182] OBJCXX obj/src/libANGLE/renderer/vulkan/angle_vulkan_backend/WindowSurfaceVkMac.o
Warning: supplying the --target arm64-apple-macos != arm64-apple-darwin argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
ninja: build stopped: subcommand failed.</pre>
</details>
</details>

--- a/pkgs/by-name/an/angle/package.nix
+++ b/pkgs/by-name/an/angle/package.nix
@@ -13,6 +13,8 @@
   wayland,
   pciutils,
   libGL,
+  apple-sdk_15,
+  xcbuild,
 }:
 let
   llvmPackages = llvmPackages_21;
@@ -29,11 +31,21 @@ let
       llvmPackages.llvm
       llvmPackages.clang
     ];
-    postBuild = ''
-      mkdir -p $out/lib/clang/${llvmMajorVersion}/lib/
-      ln -s $out/resource-root/lib/linux \
-        $out/lib/clang/${llvmMajorVersion}/lib/${triplet}
-    '';
+    postBuild =
+      if stdenv.isDarwin then
+        ''
+          mkdir -p $out/lib/clang/${llvmMajorVersion}/lib/darwin
+          ln -s $out/resource-root/lib/darwin/libclang_rt.osx.a \
+            $out/lib/clang/${llvmMajorVersion}/lib/darwin/libclang_rt.osx.a
+          ln -s $out/resource-root/lib/darwin/libclang_rt.osx.a \
+            $out/lib/clang/${llvmMajorVersion}/lib/darwin/libclang_rt.osx-${arch}.a
+        ''
+      else
+        ''
+          mkdir -p $out/lib/clang/${llvmMajorVersion}/lib/
+          ln -s $out/resource-root/lib/linux \
+            $out/lib/clang/${llvmMajorVersion}/lib/${triplet}
+        '';
   };
 in
 stdenv.mkDerivation (finalAttrs: {
@@ -51,18 +63,25 @@ stdenv.mkDerivation (finalAttrs: {
     pkg-config
     python3
     llvmPackages.bintools
+  ]
+  ++ lib.optionals stdenv.isDarwin [
+    xcbuild
   ];
 
-  buildInputs = [
-    glib
-    xorg.libxcb.dev
-    xorg.libX11.dev
-    xorg.libXext.dev
-    xorg.libXi
-    wayland.dev
-    pciutils
-    libGL
-  ];
+  buildInputs =
+    lib.optionals stdenv.isLinux [
+      glib
+      xorg.libxcb.dev
+      xorg.libX11.dev
+      xorg.libXext.dev
+      xorg.libXi
+      wayland.dev
+      pciutils
+      libGL
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      apple-sdk_15
+    ];
 
   gnFlags = [
     "is_debug=false"
@@ -73,6 +92,9 @@ stdenv.mkDerivation (finalAttrs: {
     "use_custom_libcxx=true"
     "angle_enable_swiftshader=false"
     "angle_enable_wgpu=false"
+    # On darwin during linking:
+    # clang++: error: argument unused during compilation: '-stdlib=libc++'
+    "treat_warnings_as_errors=false"
   ];
 
   patches = [
@@ -88,6 +110,12 @@ stdenv.mkDerivation (finalAttrs: {
         "_dir = \"${triplet}\"
          _suffix = \"-${arch}\""
 
+    # Don't precompile Metal shaders, because the compiler is non-free.
+    substituteInPlace src/libANGLE/renderer/metal/metal_backend.gni \
+      --replace-fail \
+        "metal_internal_shader_compilation_supported =" \
+        "metal_internal_shader_compilation_supported = false &&"
+
     cat > build/config/gclient_args.gni <<EOF
     # Generated from 'DEPS'
     checkout_angle_internal = false
@@ -95,12 +123,17 @@ stdenv.mkDerivation (finalAttrs: {
     checkout_angle_restricted_traces = false
     generate_location_tags = false
     EOF
+
+    # For sandboxed build on darwin.
+    patchShebangs build/toolchain/apple
   '';
 
   installPhase = ''
     runHook preInstallPhase
 
-    install -v -m755 -D *.so *.so.1 -t "$out/lib"
+    install -v -m755 -D \
+      *${stdenv.hostPlatform.extensions.sharedLibrary}* \
+      -t "$out/lib"
     install -v -m755 -D \
       angle_shader_translator \
       gaussian_distribution_gentables \
@@ -154,6 +187,5 @@ stdenv.mkDerivation (finalAttrs: {
     ];
     license = lib.licenses.bsd3;
     platforms = lib.platforms.linux ++ lib.platforms.darwin;
-    broken = stdenv.hostPlatform.isDarwin;
   };
 })

